tests/components/neevo/test_config_flow.py

```diff
@@ -0,0 +1,143 @@
+"""Test the Nee-Vo Tank Monitoring config flow."""
+
+from unittest.mock import patch
+
+from pyneevo import NeeVoApiInterface
+from pyneevo.errors import InvalidCredentialsError, PyNeeVoError
+
+from homeassistant.components.neevo.const import DOMAIN
+from homeassistant.config_entries import SOURCE_USER
+from homeassistant.const import CONF_EMAIL, CONF_PASSWORD
+from homeassistant.data_entry_flow import FlowResultType
+
+from tests.common import MockConfigEntry
+
+
+async def test_bad_credentials(hass):
+    """Test when provided credentials are rejected."""
+
+    result = await hass.config_entries.flow.async_init(
+        DOMAIN, context={"source": SOURCE_USER}
+    )
+    assert result["type"] == FlowResultType.FORM
+    assert result["step_id"] == SOURCE_USER
+
+    with (
+        patch(
+            "pyneevo.NeeVoApiInterface.login",
+            side_effect=InvalidCredentialsError(),
+        ),
+        patch("homeassistant.components.neevo.async_setup", return_value=True),
+        patch("homeassistant.components.neevo.async_setup_entry", return_value=True),
+    ):
+        result = await hass.config_entries.flow.async_configure(
+            result["flow_id"],
+            user_input={
+                CONF_EMAIL: "admin@localhost.com",
+                CONF_PASSWORD: "password0",
+            },
+        )
+
+        assert result["type"] == FlowResultType.FORM
+        assert result["step_id"] == "user"
+        assert result["errors"] == {
+            "base": "invalid_auth",
+        }
+
+
+async def test_generic_error_from_library(hass):
+    """Test when connection fails."""
+
+    result = await hass.config_entries.flow.async_init(
+        DOMAIN, context={"source": SOURCE_USER}
+    )
+    assert result["type"] == FlowResultType.FORM
+    assert result["step_id"] == SOURCE_USER
+
+    with (
+        patch(
+            "pyneevo.NeeVoApiInterface.login",
+            side_effect=PyNeeVoError(),
+        ),
+        patch("homeassistant.components.neevo.async_setup", return_value=True),
```

Comment:
This line seems to not be needed. Same above

Reply:
Tests fail because of socket.socket if I don't have that.

```
FAILED tests/components/neevo/test_config_flow.py::test_bad_credentials - pytest_socket.SocketBlockedError: A test tried to use socket.socket.
FAILED tests/components/neevo/test_config_flow.py::test_generic_error_from_library - pytest_socket.SocketBlockedError: A test tried to use socket.socket.
FAILED tests/components/neevo/test_config_flow.py::test_auth_worked - pytest_socket.SocketBlockedError: A test tried to use socket.socket.FAILED tests/components/neevo/test_config_flow.py::test_bad_credentials - pytest_socket.SocketBlockedError: A test tried to use socket.socket.
FAILED tests/components/neevo/test_config_flow.py::test_generic_error_from_library - pytest_socket.SocketBlockedError: A test tried to use socket.socket.
FAILED tests/components/neevo/test_config_flow.py::test_auth_worked - pytest_socket.SocketBlockedError: A test tried to use socket.socket.
```